audit/src/main/java/gov/cms/ab2d/audit/cleanup/FileDeletionServiceImpl.java

```diff
@@ -60,27 +61,111 @@ public void deleteFiles() {
+            }
+        }
+    }
+
+    void deleteExpiredJobFiles(List<Path> validFiles) {
+        // The list of jobs that were expired
+        Set<String> jobsDeleted = new HashSet<>();
+        for (Path file : validFiles) {
+
+            try {
                 // Get the JobId from the name
-                var jobUuid = new File(file.toUri()).getParentFile().getName();
+                var jobUuid = getJobFromFile(file);
                 // See if the file should be deleted
                 boolean deleted = deleteFile(file);
                 if (deleted) {
-                    // If it was deleted
+                    // If it was deleted add it to the list of jobs with deleted files
                     jobsDeleted.add(jobUuid);
                 }
+            } catch (Exception exception) {
+                log.error("failed to delete or process a regular file {}", file.toUri().toString(), exception);
             }
-            for (String job : jobsDeleted) {
-                eventLogger.log(LogManager.LogType.KINESIS, doSummary.getSummary(job));
-            }
-        } catch (IOException e) {
-            log.error("Encountered exception while trying to gather the list of files to delete", e);
+        }
+        for (String job : jobsDeleted) {
+            eventLogger.log(LogManager.LogType.KINESIS, doSummary.getSummary(job));
         }
     }
 
+    private String getJobFromFile(Path file) {
+        return new File(file.toUri()).getParentFile().getName();
+    }
+
+    /**
+     * Check whether directory contains any files
+     * @param directory directory to check which must exist
+     * @return true if a file is found in directory
+     * @throws IOException on failure to read directory
+     */
+    private boolean isEmptyDirectory(Path directory) throws IOException {
+        // Lazily look for first child
+        try (Stream<Path> children =  Files.list(directory)) {
+            return children.findAny().isEmpty();
+        }
+    }
+
+    /**
+     * Check whether a directory is associated with a job, if the directory is check that the job has completed more
+     * than x hours ago.
+     * @param directory directory to check
+     * @return true if directory belongs to a job that has finished long enough ago
+     * @throws IOException should not throw ever because getDeleteCheckTime does not hit edge case
+     */
+    private boolean isExpiredDirectory(Path directory) throws IOException {
+        var jobUuid = new File(directory.toUri()).getName();
+        var job = findJob(jobUuid, directory);
+
+        if (job == null) {
+            return false;
+        }
+
+        // Does not throw IO Exception because else clause is never hit
+        Instant deletedAt = getDeleteCheckTime(directory, job);
+        if (deletedAt == null) {
+            return false;
+        }
+
+        Instant oldestDeletableTime = calculateOldestDeletableTime();
```

Comment:
Isn't this backwards in naming?  The method is calculating the most recent boundary window of which anything older can be deleted.  It is really the youngest time, not the oldest in which deletes are warranted.

Otherwise, one would want to say something like oldestKeepMeTime.

Both the method and the variables should be renamed to something like (pick one or roll your own):
keepBoundary
deleteBoundary
youngestAgedOut
keepAnythingLater